src/cdomains/intDomain.ml

```diff
@@ -1861,193 +1861,247 @@ module Booleans = MakeBooleans (
     let falsename = "False"
   end)
 
-(* Inclusion/Exclusion sets. Go to top on arithmetic operations after ana.int.enums_max values. Joins on widen, i.e. precise integers as long as not derived from arithmetic expressions. *)
-module Enums : IkindUnawareS = struct
+(* Inclusion/Exclusion sets. Go to top on arithmetic operations (except for some easy cases, e.g. multiplication with 0). Joins on widen, i.e. precise integers as long as not derived from arithmetic expressions. *)
+module Enums : S with type int_t = BigInt.t = struct
   open Batteries
-  module I = Integers
+  module I = BigInt
   module R = Interval32 (* range for exclusion *)
 
   let range_ikind = Cil.IInt
   let size t = R.of_interval range_ikind (let a,b = Size.bits_i64 t in Int64.neg a,b)
-  type e = I.t (* element *)
-  and t = Inc of e list | Exc of e list * R.t [@@deriving to_yojson] (* inclusion/exclusion set *)
-  type int_t = int64
+  module ISet = struct
+    include Set.Make(I)
+    let is_singleton s = cardinal s = 1
+  end
+  type t = Inc of ISet.t | Exc of ISet.t * R.t (* inclusion/exclusion set *)
+
+  type int_t = BI.t
   let name () = "enums"
-  let top_range = R.of_interval range_ikind (-99L, 99L) (* Since there is no top ikind we use a range that includes both ILongLong [-63,63] and IULongLong [0,64]. Only needed for intermediate range computation on longs. Correct range is set by cast. *)
-  let bot () = Inc []
-  let top_of ik = Exc ([], size ik)
-  let top () = Exc ([], top_range)
-  let top_of ik = top ()
-  let bot_of ik = bot ()
-  let is_top x = x = top ()
+  let to_yojson x = failwith @@ "to_yojson unimplemented for " ^ (name ())
```

Comment:
Not sure if we really need this for anything anymore, but maybe `[@@deriving to_yojson]` would work here also.

Reply:
Yes, this works now when `SetDomain.Make` is used.